pkg/scclient/pool.go

```diff
@@ -0,0 +1,114 @@
+package client
+
+import (
+	"net"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var onceInit sync.Once
+var onceMonitor sync.Once
+var instance *AddressPool
+
+const (
+	available               string = "available"
+	unavailable             string = "unavailable"
+	defaultCheckSCIInterval        = 25 // default sc instance health check interval in second
+)
+
+// AddressPool registry address pool
+type AddressPool struct {
+	addressMap map[string]string
+	status     map[string]string
+	mutex      sync.RWMutex
+}
+
+// GetInstance Get registry pool instance
+func GetInstance() *AddressPool {
+	onceInit.Do(func() {
+		instance = &AddressPool{
+			addressMap: make(map[string]string),
+			status:     make(map[string]string),
+		}
+	})
+	return instance
+}
+
+// SetAddress set addresses to pool
+func (p *AddressPool) SetAddress(addresses []string) {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+	p.addressMap = make(map[string]string)
+	for _, v := range addresses {
+		p.addressMap[v] = v
+	}
+}
+
+// GetAvailableAddress Get an available address from pool by roundrobin
+func (p *AddressPool) GetAvailableAddress() string {
+	p.mutex.Lock()
```

Comment:
you only need to use r lock, not w lock

Reply:
in SetAddress need w lock?

Reply:
> you only need to use r lock, not w lock

in SetAddress need w lock?

Reply:
yes，read and write， it depends which operation you do to a map.

rlock allows concurrency get from map

Reply:
done

Reply:
> yes，read or write， it depends which operation you do to a map.
> 
> rlock allows concurrency get from map

ok.